server/src/autocomplete/node-completion.ts

```diff
@@ -1,15 +1,19 @@
 import { CompletionItem, CompletionItemKind } from 'vscode-languageserver'
-import { Node, Body, Method, Singleton } from 'wollok-ts'
+import { Node, Body, Method, Singleton, Module, Environment, Package } from 'wollok-ts'
+import { is, match, when } from 'wollok-ts/dist/extensions'
 import { fieldCompletionItem, parameterCompletionItem, singletonCompletionItem } from './autocomplete'
 
 export const completionsForNode = (node: Node): CompletionItem[] => {
-  switch (node.kind) {
-    case 'Environment': return []
-    case 'Package': return completePackage()
-    case 'Singleton': return completeSingleton()
-    case 'Body': return completeBody(node)
-    case 'Method': return completeMethod(node)
-    default: return completeForParent(node)
+  try{
+    return match(node)(
+      when(Environment)(_ => []),
+      when(Package)(completePackage),
+      when(Singleton)(completeSingleton),
+      when(Body)(completeBody),
+      when(Method)(completeMethod)
+    )
+  } catch {
+    return completeForParent(node)
```

Comment:
Hace falta este `try - catch`? Es porque el `match` cuando no encuentra ninguno explota o por otra cosa que puede pasar?

Tal vez estaría bueno tener algo onda 
```wollok
match(node)(
   when(Environment)(_ => []),
   ...,
   otherwise(completeForParent)
)
```

Reply:
Esto estaria buenisimo, y que si no tiene un otherwise ahi si que lance una excepcion